src/paragraph/Paragraph.tsx

```diff
@@ -7,59 +7,68 @@ import { StyledParagraph } from './Paragraph.style'
 
 const DEFAULT_MAX_CHAR_SIZE = 180
 
-export type ParagraphProps = Readonly<{
+type ParagraphBaseProps = Readonly<{
   className?: string
-  children: string
-  isExpandable?: boolean
-  expandLabel?: string
   itemProp?: string
 }>
 
-type ParagraphState = {
-  isExpanded: boolean
+export type ParagraphSimpleProps = ParagraphBaseProps &
+  Readonly<{
+    children: string
+  }>
+
+export type ParagraphExpandableProps = ParagraphBaseProps &
+  Readonly<{
+    isExpandable: true
+    expandLabel: string
+    children: string
+  }>
+
+export type ParagraphProps = ParagraphSimpleProps | ParagraphExpandableProps
+
+const truncateText = (text: string) => `${text.substring(0, DEFAULT_MAX_CHAR_SIZE)}…`
```

Comment:
Since you are refactoring this component, could you please review #a11y with @nickoola and behavior with @Tuxosaurus  for SEO.

Thanks :)

Reply:
I think this PR is not a refacto but the ability to use inline node for the text part.
The expand mechanism for a11y can be improved but require discussion with Pixar as not consistent between components. We need to convince designers to keep the expand button instead of hide it after click.
Here is the corresponding jira for this task [BBC-12577](https://blablacar.atlassian.net/browse/BBC-12577) 
So for A11y no change asked on this PR.